### PR TITLE
Enhancement: Enabled feature to hibernate & unhibernate the external helm apps

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

These changes will enable the feature to hibernate & unhibernate the external helm apps.
UX - https://www.figma.com/file/j7cqlBdhQ47Bo10TLfamGl/Helm-Apps?node-id=755%3A3237

Fixes # https://github.com/devtron-labs/devtron/issues/1423

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/appDetails/sourceInfo/scaleWorkloads/ScaleWorkloadsModal.component.tsx [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    const handleWorkloadSelection = (workloadKey: string, isActiveWorkloadsTab: boolean): void => {
+        const _workloadsList = isActiveWorkloadsTab ? workloadsToScaleDown : workloadsToRestore
+        const _setWorkloadsList = isActiveWorkloadsTab ? setWorkloadsToScaleDown : setWorkloadsToRestore
+
+        for (let [key, value] of _workloadsList) {
+            if (workloadKey === `${value.kind}/${value.name}`) {
+                value.value = !value.isChecked ? 'CHECKED' : 'INTERMEDIATE'
+                value.isChecked = !value.isChecked
+            }
+            _workloadsList.set(key, value)
```
Comment (vivek-devtron) [inline src/components/v2/appDetails/sourceInfo/scaleWorkloads/ScaleWorkloadsModal.component.tsx:218]: Can we move this in the if block?

File: src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ function EnvironmentStatusComponent({appStreamData}:{appStreamData: any}) {
     const [appDetails] = useSharedState(IndexStore.getAppDetails(), IndexStore.getAppDetailsObservable());
     const [showAppStatusDetail, setShowAppStatusDetail] = useState(false);
     const status = appDetails.resourceTree?.status || '';
+    const isHibernated = status.toLowerCase().includes('hibernated')
```
Comment (manish-agrawal-ai) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx:19]: includes or equals?

Comment (sohel-rp) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/EnvironmentStatus.component.tsx]: It'll work with `includes` as we have `hibernated` in both the statuses and will handle both scenarios. However, as discussed, I've put the specific checks just to make sure we're showing it only for those specific statuses. :)

File: src/components/v2/appDetails/sourceInfo/scaleWorkloads/ScaleWorkloadsModal.component.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/v2/appDetails/sourceInfo/scaleWorkloads/ScaleWorkloadsModal.component.tsx]: Done. Removed the for-loop & if check itself. As we already have the key to fetch the particular value from the map.

**Review by vivek-devtron — APPROVED** [review]

**Review by manish-agrawal-ai — APPROVED** [review]

**Pull request merged**